--- atom.h	2016-03-29 14:46:40.525473896 -0400
+++ neww	2016-03-29 14:49:29.431748896 -0400
@@ -45,6 +45,8 @@
 
   // per-atom arrays
   // customize by adding new array
+  bigint *gid;
+  double **xeq;
 
   tagint *tag;
   int *type,*mask;
@@ -127,6 +129,7 @@
 
   // atom style and per-atom array existence flags
   // customize by adding new flag
+  int gfmd_flag;
 
   int sphere_flag,ellipsoid_flag,line_flag,tri_flag,body_flag;
   int peri_flag,electron_flag;
