diff --git a/src/atom.cpp b/src/atom.cpp
index ecb82993c..6abbf2ea6 100644
--- a/src/atom.cpp
+++ b/src/atom.cpp
@@ -105,6 +105,9 @@ Atom::Atom(LAMMPS *lmp) : Pointers(lmp)
   // --------------------------------------------------------------------
   // 1st customization section: customize by adding new per-atom variables
 
+  gid = nullptr;
+  xeq = nullptr;
+
   tag = nullptr;
   type = mask = nullptr;
   image = nullptr;
@@ -274,6 +277,9 @@ Atom::~Atom()
   memory->destroy(next);
   memory->destroy(permute);
 
+  memory->destroy(gid);
+  memory->destroy(xeq);
+
   memory->destroy(tag);
   memory->destroy(type);
   memory->destroy(mask);
@@ -599,6 +605,8 @@ void Atom::set_atomflag_defaults()
   // 3rd customization section: customize by adding new flag
   // identical list as 2nd customization in atom.h
 
+  gfmd_flag = 0;
+
   sphere_flag = ellipsoid_flag = line_flag = tri_flag = body_flag = 0;
   peri_flag = electron_flag = 0;
   wavepacket_flag = sph_flag = 0;
