diff --git a/src/atom.h b/src/atom.h
index 70e064c0ff..4a499a22e2 100644
--- a/src/atom.h
+++ b/src/atom.h
@@ -174,6 +174,12 @@ class Atom : protected Pointers {
 
   double *area, *ed, *em, *epsilon, *curvature, *q_unscaled;
 
+  // USER-GFMD package
+
+  int **gid;
+  int *gflag;
+  double **xeq;
+
   // end of customization section
   // --------------------------------------------------------------------
 
@@ -212,6 +218,10 @@ class Atom : protected Pointers {
 
   int nspecial15_flag;
 
+  // USER-GFMD package
+
+  int gfmd_flag;
+
   // Peridynamics scale factor, used by dump cfg
 
   double pdscale;
