diff --git a/src/atom.cpp b/src/atom.cpp
index 3d843cbbbc..ac1e97ec09 100644
--- a/src/atom.cpp
+++ b/src/atom.cpp
@@ -216,6 +216,12 @@ Atom::Atom(LAMMPS *_lmp) : Pointers(_lmp)
 
   area = ed = em = epsilon = curvature = q_unscaled = nullptr;
 
+  // USER-GFMD package
+
+  gid = nullptr;
+  gflag = nullptr;
+  xeq = nullptr;
+
   // end of customization section
   // --------------------------------------------------------------------
 
@@ -565,6 +571,12 @@ void Atom::peratom_create()
   add_peratom("curvature",&curvature,DOUBLE,0);
   add_peratom("q_unscaled",&q_unscaled,DOUBLE,0);
 
+  // USER-GFMD package
+
+  add_peratom("gid",&gid,INT,3);
+  add_peratom("gflag",&gflag,INT,0);
+  add_peratom("xeq",&xeq,DOUBLE,3);
+
   // end of customization section
   // --------------------------------------------------------------------
 }
@@ -646,6 +658,7 @@ void Atom::set_atomflag_defaults()
   contact_radius_flag = smd_data_9_flag = smd_stress_flag = 0;
   eff_plastic_strain_flag = eff_plastic_strain_rate_flag = 0;
   nspecial15_flag = 0;
+  gfmd_flag = 0;
 
   pdscale = 1.0;
 }
@@ -2952,6 +2965,11 @@ void *Atom::extract(const char *name)
   if (strcmp(name,"curvature") == 0) return (void *) curvature;
   if (strcmp(name,"q_unscaled") == 0) return (void *) q_unscaled;
 
+  // USER-GFMD package
+  if (strcmp(name,"gid") == 0) return (void *) gid;
+  if (strcmp(name,"gflag") == 0) return (void *) gflag;
+  if (strcmp(name,"xeq") == 0) return (void *) xeq;
+
   // end of customization section
   // --------------------------------------------------------------------
 
@@ -3074,6 +3092,11 @@ int Atom::extract_datatype(const char *name)
   if (strcmp(name,"curvature") == 0) return LAMMPS_DOUBLE;
   if (strcmp(name,"q_unscaled") == 0) return LAMMPS_DOUBLE;
 
+  // USER-GFMD package
+  if (strcmp(name,"gid") == 0) return LAMMPS_INT;
+  if (strcmp(name,"gflag") == 0) return LAMMPS_INT;
+  if (strcmp(name,"xeq") == 0) return LAMMPS_DOUBLE_2D;
+
   // end of customization section
   // --------------------------------------------------------------------
 
