diff --git a/src/pair.h b/src/pair.h
index 4f4d66ed97..82c64eb13d 100644
--- a/src/pair.h
+++ b/src/pair.h
@@ -245,7 +245,9 @@ class Pair : protected Pointers {
   double tabinner;              // inner cutoff for Coulomb table
   double tabinner_disp;         // inner cutoff for dispersion table
 
- protected:
+// Next line commented out by USER-GFMD.
+// protected:
+ public:
   // for mapping of elements to atom types and parameters
   // mostly used for manybody potentials
   int nelements;        // # of unique elements
